src/runtime/api-channel.ts

```diff
@@ -203,17 +203,35 @@ class ThingMapper {
   }
 }
 
+export abstract class Port {
+  abstract close(): void;
+  abstract postMessage(msg): void;
+  abstract setMessageCallback(callback): void;
```

Comment:
Can one use one of the types in `hot.ts` for callback?  Maybe Runnable?

Reply:
I tried to add Runnable, but then it's incompatible with MessagePort (the code was refactor, so `_port` is now `MessagePort | Port`, i'm not sure why.